### PR TITLE
[masquerade-binding] Set the pod iface mtu on the bridge

### PR DESCRIPTION
**Pull request by AlonaKaplan**
**What this PR does / why we need it**:
Since on masquerade binding the bridge is not the master of pod interface,
it doesn't automatically get the mtu of the interface.
This PR copies the mtu from the pod interface and sets it on the bridge.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Masquerade binding - set the virt-launcher pod interface MTU on the bridge.
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/network/podinterface.go
Comment (dankenigsberg) [inline pkg/virt-launcher/virtwrap/network/podinterface.go:854]: I love one liners!

**Comment by EdDev** [comment]
> This PR copies the mtu from the pod interface and sets it on the bridge.

If I remember correctly, the mtu if the bridge management interface is set automatically per the lowest connected port to it.
Isn't the tap device set with an appropriate mtu? or it is and what I remember is incorrect?

**Comment by AlonaKaplan** [comment]
> > This PR copies the mtu from the pod interface and sets it on the bridge.
> 
> If I remember correctly, the mtu if the bridge management interface is set automatically per the lowest connected port to it.
> Isn't the tap device set with an appropriate mtu? or it is and what I remember is incorrect?

I think it is becuase k6t-eth0-nic has lower mtu (the default one).

File: pkg/virt-launcher/virtwrap/network/podinterface.go
Comment (AlonaKaplan) [reply pkg/virt-launcher/virtwrap/network/podinterface.go:855]: Well two lines. Just to be on the safe side. Although from my experiments when the master has an explicit MTU it doesn't take the MTU from a slave, even if the slave has lower value.

File: tests/vmi_networking_test.go
Comment (EdDev) [inline tests/vmi_networking_test.go:808]: Is this a debugging leftover?

File: tests/vmi_networking_test.go
Comment (EdDev) [inline tests/vmi_networking_test.go:824]: Leftover?

File: tests/vmi_networking_test.go
Comment (EdDev) [inline tests/vmi_networking_test.go:834]: I think you can do this in parallel, i.e. start the creation of both and then wait in sequence.

File: tests/vmi_networking_test.go
Comment (EdDev) [inline tests/vmi_networking_test.go:864]: It is safer to use `vmi.Namespace` here.

File: tests/vmi_networking_test.go
Comment (EdDev) [inline tests/vmi_networking_test.go:884]: How about using the same command as you did for the in-pod case? I think it is more "direct".

File: tests/vmi_networking_test.go
Comment (EdDev) [inline tests/vmi_networking_test.go:797]: Unless  missed something, this test is using the default MTU which should have no issues even today. The problem occurs when the specified MTU is bigger than the default dummy one (1500?).

However, I am not sure if you can set a bigger mtu (we may need to adjust/check kubevirtci for this).

Comment (AlonaKaplan) [reply tests/vmi_networking_test.go:797]: Yes, the test is using the mtu the cni provides. kubevirt doesn't have api to control the mtu.
So yes, if we want jumbo frames test we need to whether update cni on our existing cluster or create a new cluster.

However, the reason to the bug is the bridge got its mtu from k6t-eth0-nic and not from the tap device.
The existing test verifies the k6t-eth0 bridge, k6t-eth0-nic and the vnic have the same correct mtu. Due to the urgency, I think is the best we can have atm. The test will fail on existing cluster without the fix in the first commit of the PR (so it passes the "make sure to start with a failing test" criteria:)).

File: tests/vmi_networking_test.go
Comment (AlonaKaplan) [reply tests/vmi_networking_test.go:808]: It was copied from the bridge mtu test. But indeed pretty redundant. I will remove.

File: tests/vmi_networking_test.go
Comment (AlonaKaplan) [reply tests/vmi_networking_test.go:824]: yes, done

File: tests/vmi_networking_test.go
Comment (AlonaKaplan) [reply tests/vmi_networking_test.go:864]: done

File: tests/vmi_networking_test.go
Comment (AlonaKaplan) [reply tests/vmi_networking_test.go:884]: It was also copied from the bridge mtu test. I will try.

File: tests/vmi_networking_test.go
Comment (AlonaKaplan) [reply tests/vmi_networking_test.go:825]: Done, I hope I understood your intention.

File: tests/vmi_networking_test.go
Comment (EdDev) [inline tests/vmi_networking_test.go:841]: You understood it correctly but I think this is an overkill.
Waiting in sequence is enough, no need for goroutine here.
After you wait for the 1st, the other is either already ready or we need a few more seconds. If everything goes right, we will not wait more than needed.

Except that, using a goroutine is a problem when the helper functions assert, ginkgo do not like it and cause problems. It requires special handling.

**Review by EdDev — APPROVED** [review]
Thank you!

It still needs to be rebased after #4156 is merged, so adding a hold.

/hold

File: pkg/virt-launcher/virtwrap/network/podinterface.go
Comment (vladikr) [reply pkg/virt-launcher/virtwrap/network/podinterface.go:855]: Does this change solve the issue?
IIRC, the bridge will override the set MTU with the lowest MTU on the enslaved devices, isn't it?

Comment (vladikr) [reply pkg/virt-launcher/virtwrap/network/podinterface.go:855]: ah, that's on masquerade...

**Comment by vladikr** [comment]
@AlonaKaplan Perhaps you could update the title to reflect that this changes the masquerade binding bridge and add a release note.

**Comment by AlonaKaplan** [comment]
> @AlonaKaplan Perhaps you could update the title to reflect that this changes the masquerade binding bridge and add a release note.

Thanks for the review! Done.

**Comment by AlonaKaplan** [comment]
puhsed - rebase

**Review by EdDev — APPROVED** [review]

**Pull request merged**
